src/ctrlc.rs

```diff
@@ -0,0 +1,86 @@
+use console::Term;
+use once_cell::sync::Lazy;
+use signal_hook::{
+    consts::SIGINT,
+    iterator::{Handle, Signals},
+};
+use std::{
+    io::Error,
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Mutex, RwLock,
+    },
+    thread,
+    time::Duration,
+};
+
+static MUTEX: Mutex<()> = Mutex::new(());
+static INIT: AtomicBool = AtomicBool::new(false);
+static HANDLE: Lazy<RwLock<Option<CtrlcHandle>>> = Lazy::new(|| RwLock::new(None));
```

Comment:
using `Lazy::new(Default::default())` does not seem to work since `Handle` does not implement a default